device/core/src/device_client.ts

```diff
@@ -96,7 +96,7 @@ export class Client extends InternalClient {
       if (err && this._retryPolicy.shouldRetry(err)) {
         debugErrors('reconnect policy specifies a reconnect on error');
         /*Codes_SRS_NODE_DEVICE_CLIENT_16_097: [If the transport emits a `disconnect` event while the client is subscribed to c2d messages the retry policy shall be used to reconnect and re-enable the feature using the transport `enableC2D` method.]*/
-        if (this._c2dFeature) {
+        if (this._userRegisteredC2dListener) {
```

Comment:
We want to re-enable C2D if the user registered a listener.
Before, we were enabling if the user registered a listener AND the transport successfully enabled C2D prior.